R/Reporter.R

```diff
@@ -433,14 +434,22 @@ Reporter <- R6::R6Class( # nolint: object_name_linter.
     },
     #' @description Get the `Reporter` template
     #' @return a template `function`.
-    get_template = function() private$template
+    get_template = function() private$template,
+    #' @description Compare two Reporter objects for equality.
+    #' @param other (`Reporter`) another `Reporter` object to compare with.
+    #' @return `logical(1)` indicating whether the two `Reporter` objects are equal.
+    equals = function(other) {
+      checkmate::assert_class(other, "Reporter")
+      identical(self$get_id(), other$get_id()) &&
+        identical(self$get_metadata(), other$get_metadata()) &&
+        identical(unname(self$get_cards()), unname(other$get_cards()))
+    }
```

Comment:
Can you avoid creating a public method to be used only in tests? Method is based on a public methods so it is easy to write a private util in`helper-Reporter.R`